src/extensions/default/InlineBorderRadiusEditor/BorderRadiusEditor.js

```diff
@@ -0,0 +1,220 @@
+        var $allCornersArea = self.$element.find("#allCornersArea");
+        var $allCornerButton = self.$element.find("#allCorners");
+
+        function toggleCornerOption($showElement, $hideElement) {
+            $showElement.show();
+            $hideElement.hide();
+            self.allCorners = $showElement === $allCornersArea;
+            self.onChange();
+        }
+
+        $allCornerButton.on("click", function() {
+            $allCornerButton.addClass("selected");
+            $individualCornerButton.removeClass("selected");
+            toggleCornerOption($allCornersArea, $individualCornerArea);
+        });
+        $individualCornerButton.on("click", function() {
+            $allCornerButton.removeClass("selected");
+            $individualCornerButton.addClass("selected");
+            toggleCornerOption($individualCornerArea, $allCornersArea);
+        });
+
+        // initialize individual corner editing to be disabled if allCorner is set to true
+        if(self.allCorners){
+            $allCornerButton.trigger("click");
+        } else {
+            $individualCornerButton.trigger("click");
+        }
+    }
+
+    BorderRadiusEditor.prototype.updateValues = function(valueString) {
+        var values = getIndividualValues(valueString);
+        var numOfValues = values.length;
+        var firstValue = values[0];
+        var secondValue = firstValue;
+        var thirdValue = firstValue;
+        var fourthValue = firstValue;
+
+        this.allCorners = values.length === 1;
+
+        if (!this.allCorners) {
+            if (numOfValues > 1) {
+                secondValue = values[1];
+            }
+            if (numOfValues > 2) {
+                thirdValue = values[2];
+            }
+            if (numOfValues > 3) {
+                fourthValue = values[3];
+            } else if (numOfValues > 1 && numOfValues < 4) {
+                fourthValue = values[1];
+            }
+        }
+
+        this.topLeft = new BorderRadiusValue(
+            this.$element,
+            "top-left",
+            firstValue.num,
+            firstValue.unit,
+            this.onChange
+        );
+        this.topRight = new BorderRadiusValue(
+            this.$element,
+            "top-right",
+            secondValue.num,
+            secondValue.unit,
+            this.onChange
+        );
+        this.bottomLeft = new BorderRadiusValue(
+            this.$element,
+            "bottom-left",
+            thirdValue.num,
+            thirdValue.unit,
+            this.onChange
+        );
+        this.bottomRight = new BorderRadiusValue(
+            this.$element,
+            "bottom-right",
+            fourthValue.num,
+            fourthValue.unit,
+            this.onChange
+        );
+        this.allCorner = new BorderRadiusValue(
+            this.$element,
+            "all-corner",
+            firstValue.num,
+            firstValue.unit,
+            this.onChange
+        );
+    };
```

Comment:
before ending the function, call `onChange` so that we correctly update the values in the UI.